akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/AccountExample6.scala

```diff
@@ -0,0 +1,120 @@
+    case class Deposit(amount: Double) extends AccountCommand
+    case class Withdraw(amount: Double) extends AccountCommand
+    case object CloseAccount extends AccountCommand
+
+    sealed trait AccountEvent
+    case object AccountCreated extends AccountEvent
+    case class Deposited(amount: Double) extends AccountEvent
+    case class Withdrawn(amount: Double) extends AccountEvent
+    case object AccountClosed extends AccountEvent
+
+
+    type Effect = akka.persistence.typed.scaladsl.Effect[AccountEvent, Account]
+
+    sealed trait Account {
+      def applyCommand(cmd: AccountCommand): Effect
+      def applyEvent(event: AccountEvent): Account
+    }
+
+
+    case object EmptyAccount extends Account {
+
+      override def applyCommand(cmd: AccountCommand): Effect =
+        cmd match {
+          case CreateAccount ⇒ Effect.persist(AccountCreated)
+          case _             ⇒ Effect.unhandled
+        }
+
+      override def applyEvent(event: AccountEvent): Account = event match {
+        case AccountCreated ⇒ OpenedAccount(0.0)
+        case _              ⇒ throw new IllegalStateException(s"unexpected event [$event] in state [EmptyAccount]")
+      }
+
+    }
+
+    case class OpenedAccount(balance: Double) extends Account {
+
+      override def applyCommand(cmd: AccountCommand): Effect =
+        cmd match {
+          case Deposit(amount) ⇒ Effect.persist(Deposited(amount))
+
+          case Withdraw(amount) ⇒
+            if ((this.balance - amount) < 0.0)
+              Effect.unhandled // TODO replies are missing in this example
+            else {
+              Effect
+                .persist(Withdrawn(amount))
+                .thenRun {
+                  case OpenedAccount(balance) ⇒
+                    // do some side-effect using balance
+                    println(balance)
+                  case _ ⇒ throw new IllegalStateException
+                }
+            }
+          case CloseAccount if this.balance == 0.0 ⇒
+            Effect.persist(AccountClosed)
+
+          case CloseAccount ⇒
+            Effect.unhandled
+        }
+
+
+      override def applyEvent(event: AccountEvent): Account = event match {
+        case Deposited(amount) ⇒ copy(balance = balance + amount)
+        case Withdrawn(amount) ⇒ copy(balance = balance - amount)
+        case AccountClosed     ⇒ ClosedAccount
+        case _                 ⇒ throw new IllegalStateException(s"unexpected event [$event] in state [OpenedAccount]")
+      }
+
+    }
+
+    case object ClosedAccount extends Account {
+
+      override def applyCommand(cmd: AccountCommand): Effect =
+        _ ⇒ Effect.unhandled
+
+      override def applyEvent(event: AccountEvent): Account =
+        throw new IllegalStateException(s"unexpected event [$event] in state [ClosedAccount]")
+
+    }
+
+
+    def behavior(accountNumber: String): Behavior[AccountEntity.AccountCommand] = {
+      PersistentBehaviors6.receive(
```

Comment:
the name choice of `receive` came from plain `Behaviors.receive`, but I have started to think that it might read better with `PersistentBehavior.apply`, also since it's defining more than just receiving messages (commads)